Src/main.c

```diff
@@ -118,76 +126,82 @@ int main(void)
 
   /* USER CODE BEGIN SysInit */
   /* Initialize LEDs */
-  BSP_LED_Init(LED9);
-  BSP_LED_Init(LED10);
+  //BSP_LED_Init(LED9);
+  //BSP_LED_Init(LED10);
   /* USER CODE END SysInit */
 
   /* Initialize all configured peripherals */
+
   MX_ICACHE_Init();
+
   /* USER CODE BEGIN 2 */
 
   /* USER CODE END 2 */
 
   /* Init scheduler */
   osKernelInitialize();
 
-  /* USER CODE BEGIN RTOS_MUTEX */
-  printf("Main func\n");
-
-  /* USER CODE END RTOS_MUTEX */
-
-  /* USER CODE BEGIN RTOS_SEMAPHORES */
-
-  /* USER CODE END RTOS_SEMAPHORES */
-
-  /* USER CODE BEGIN RTOS_TIMERS */
-
-  /* USER CODE END RTOS_TIMERS */
-
-  /* USER CODE BEGIN RTOS_QUEUES */
-
-  /* USER CODE END RTOS_QUEUES */
-
-  /* Create the thread(s) */
-  /* creation of THREAD1 */
-  //THREAD1Handle = osThreadNew(LED_Thread1, NULL, &THREAD1_attributes);
-
-
-    status = xTaskCreate(LED_Thread1, "Task1", 500, &s1, 2, &task1_handle);
-    status1 = xTaskCreate(LED_Thread2, "Task2", 500, "Task-2 is running", 2, &task2_handle);
-    configASSERT(status == pdPASS);
-    configASSERT(status1 == pdPASS);
-
-    printf("Main Num :%d\n",s1.num);
-
-
-  /* creation of THREAD2 */
-  //THREAD2Handle = osThreadNew(LED_Thread2, NULL, &THREAD2_attributes);
-
-  /* USER CODE BEGIN RTOS_THREADS */
-  /* USER CODE END RTOS_THREADS */
-
-  /* USER CODE BEGIN RTOS_EVENTS */
-  /* add events, ... */
-  /* USER CODE END RTOS_EVENTS */
+    EXTI13_IRQHandler_Config();
+
+    MX_USART1_UART_Init();
+    USART1->CR1 |= USART_CR1_RXNEIE;
+    HAL_UART_Transmit(&huart1, msg, sizeof(msg), 1000);
+    while(!(USART1->ISR & USART_ISR_TXE));
+    USART1->TDR='A';
+
+    BaseType_t status;
+    xPointerQueue = xQueueCreate( QUEUE_LENGTH, sizeof( char * ) );
+    if( xPointerQueue != NULL )
+       {
+
+    	   status = xTaskCreate(vSenderTask, "Task1", 500, NULL, 0, &task1_handle);
+    	   configASSERT(status == pdPASS);
+       vTaskStartScheduler();
+       }
+       else
+       {
+       /* The queue could not be created. */
+    	   printf("Queue not created\n");
+       }
 
   /* Start scheduler */
-  osKernelStart();
-  printf("Main Num :%d\n",s1.num);
+  //osKernelStart();
 
-  /* We should never get here as control is now taken by the scheduler */
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+   //HAL_UART_Receive_IT(&huart1, &data, 1);
+
   while (1)
   {
     /* USER CODE END WHILE */
+	  /* USER CODE BEGIN 3 */
+/*if(flag==1)
+{
+   while(!(USART1->ISR & USART_ISR_TXE));
+   {
+	   USART1->TDR=data;
+	              flag=0;
+   }
+}*/
+
+
+	   /* if(rear>-1 && front<=rear)
```

Comment:
It is good to keep multiple logic implementation in code. 
However, instead of commenting it, enable/disable it by using macros.
See example of assert used in this file. It use USE_FULL_ASSERT
#ifdef  USE_FULL_ASSERT
// code here
#endif

So you can make it disable required lines as below,
#ifdef UART_ECHO_EX_USING_C_ARRAY
// your code here
#elif UART_ECHO_EX_USING_FREERTOS_QUEUE
// your code here
#endif

So whenever you want to test it with any logic, just define that macro in header/project property file.